### PR TITLE
feat: ALBERT on SQuAD 2.0 example

### PR DESCRIPTION
**Pull request by armandmcqueen**
## Description

Add an example of training ALBERT on SQuAD 2.0.

## Test Plan

We've ran many convergence tests. There was some minor code cleanup since the last convergence test so I am running another experiment to confirm that it trains as expected. 1 GPU training is fine. 8 GPU training is starting up.

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.

### CODE REVIEW

File: examples/nlp/albert_squad_pytorch/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+## Files
+* **model_def.py**: The core code for the model. This includes building and compiling the model.
+* **data.py**: The data loading and preparation code for the model.
+* **constants.py**: Constant references to models able to run on SQuAD.
+* **startup-hook.sh**: Additional dependencies that Determined will automatically install into each container for this experiment.
+
+### Configuration Files
+* **const.yaml**: Train the model on 1 GPU.
+* **distributed_8gpu.yaml**: Train the model on 8 GPUs (distributed training) while maintaining the same accuracy
```
Comment (naren-determined) [inline examples/nlp/albert_squad_pytorch/README.md:15]: Nit: Period at the end of this sentence (for uniformity)

File: examples/nlp/albert_squad_pytorch/README.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+| 8    | 15.8               | 85.76       | 88.87 |
+| 64   | 92.75              | 86.24       | 89.06 |
+
+
+
+### Caching
+
+Extracting features from the dataset is quite time-consuming so this code will cache the extracted features to a file and not re-extract the features if that file is present. With containers, files that are saved to the container's file system are deleted when the container closes, so to reuse the cache file across experiments, you will need to set up a `bind_mount` in the experiment configuration, which allows the container to write to the host machine's file system.  
+
+This caching works when you are running repeated experiments with the same agents, but in a cloud environment when you want to shut down VMs when they aren't in use, the cache will be emprt on any newly created VMs. To avoid this, you can have the cloud VMs use a network attached filesystem (e.g. EFS or FSx for Lustre on AWS) and bind mount a directory on the filesystem (for more details, see [our docs](https://docs.determined.ai/latest/tutorials/data-access.html#distributed-file-system))
```
Comment (naren-determined) [inline examples/nlp/albert_squad_pytorch/README.md:37]: typo: emprt --> "cache will be empty"

File: examples/nlp/albert_squad_pytorch/const.yaml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    records: 80000
+data:
+    pretrained_model_name: "albert-xxlarge-v2"
+    use_bind_mount: True
+    bind_mount_path: /mnt/data
+    task: "SQuAD2.0"  # SQuaD 2.0 has 132198 example.
+entrypoint: model_def:AlbertSQuADPyTorch
+optimizations:
+    aggregation_frequency: 24
+max_restarts: 0
```
Comment (naren-determined) [inline examples/nlp/albert_squad_pytorch/const.yaml:38]: We can probably remove this field entirely from the specification.

File: examples/nlp/albert_squad_pytorch/const.yaml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+bind_mounts:
```
Comment (naren-determined) [inline examples/nlp/albert_squad_pytorch/const.yaml:39]: Just confirming here that there is nothing the user has to do with respect to bind mounts and this is already handled by just running the `det e create` command?

File: examples/nlp/albert_squad_pytorch/data.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
+from transformers.data.processors.squad import SquadV1Processor, SquadV2Processor, SquadFeatures
+from transformers import squad_convert_examples_to_features
+import urllib.request
+import os
+import torch
+
+from pathlib import Path
+
+
```
Comment (naren-determined) [inline examples/nlp/albert_squad_pytorch/data.py:9]: Nit: Extra empty lines can be removed

**Review by naren-determined — CHANGES_REQUESTED** [review]
Can you also add this example to the top-level README as well as to `docs/examples.txt`. Also, can you follow [this](https://docs.google.com/document/d/1G5pKRllMnNXnLQKYlHGMjOfFz66OHUcGRVnSLVMs8WQ/edit) for writing tests?

File: examples/nlp/albert_squad_pytorch/const.yaml (hunk 4 above)
Comment (armandmcqueen) [reply examples/nlp/albert_squad_pytorch/const.yaml:38]: Hmm, I guess for a normal user's setup, this won't work correctly because the bind mounted directory (`/home/ubuntu/dtrain-fsx/`) won't exist.

Extracting the features takes 30+ minutes, so we really want to cache it by default. I changed it so that by default we bind mount to `/tmp` because that should always exist

**Review by naren-determined — APPROVED** [review]
In the "To Run" section, if you can add whatever needs to be done w.r.t. `bind_mounts` that would be great. Otherwise looks good to 🚢
